src/ui/molecules/stepper/selector/stepper.selector.ts

```diff
@@ -0,0 +1,76 @@
+import type { StepperState } from 'hook/useStepper'
+import { OrderedMap } from 'immutable'
+import type { DeepReadonly } from 'superTypes'
+import type { Step, StepId, StepIndex, StepStatus } from 'ui/index'
+
+/**
+ * Get the status of the steps.
+ * @param steps The step array.
+ * @returns The map of the status of the step.
+ */
+export const getStatusFromSteps = (steps: DeepReadonly<Step[]>): OrderedMap<StepId, StepStatus> =>
+  OrderedMap<StepId, StepStatus>(
+    steps.map((step: DeepReadonly<Step>) => [step.id, step.status ?? 'uncompleted'])
+  )
+
+/**
+ * Retrieve the status of the step by its ID.
+ * @param id The ID of the step.
+ * @param state The React state of the Stepper
+ * @returns The step status.
+ */
+export const getStepStatusByStepId = (id: StepId, state: DeepReadonly<StepperState>): StepStatus =>
+  state.stepStatus.get(id) ?? 'uncompleted'
+
+/**
+ * Update the status of the step thank to the Stepper React state.
+ * @param steps The step array.
+ * @param state The React state of the Stepper.
+ * @returns a step array with updated status.
+ */
+export const getStepsWithUpdatedStatus = (
```

Comment:
IMHO, it's more about combining original steps with their new statuses, maybe we could enhance the global comprehension of the method.